app/javascript/controllers/maps_controller.js

```diff
@@ -0,0 +1,85 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = ["map", "field", "getCurrentLocation"]
+  static values = { latitud: Number,
+                    longitud: Number}
+
+  connect() {
+    if (typeof (google) != "undefined") {
+      this.initializeMap()
+    }
+  }
+
+  initializeMap() {
+    this.map()
+    this.marker()
+    this.autocomplete()
+  }
+
+  map() {
+    if (this._map == undefined) {
+      this._map = new google.maps.Map(this.mapTarget, {
+        center: new google.maps.LatLng(
+          this.latitudValue,
+          this.longitudValue
+        ),
+        zoom: 6
+      })
+    }
+    return this._map
+  }
+
+  marker() {
+    if (this._marker == undefined) {
+      this._marker = new google.maps.Marker({
+        map: this.map(),
+        anchorPoint: new google.maps.Point(0, 0)
+      })
+      let mapLocation = {
+        lat: parseFloat(this.latitudValue),
+        lng: parseFloat(this.longitudValue)
+      }
+      this._marker.setPosition(mapLocation)
+      this._marker.setVisible(true)
+    }
+    return this._marker
+  }
+
+  autocomplete() {
+    if (this._autocomplete == undefined) {
+      this._autocomplete = new google.maps.places.Autocomplete(this.fieldTarget)
+      this._autocomplete.bindTo('bounds', this.map())
+      this._autocomplete.setFields(['address_components', 'geometry', 'icon', 'name'])
+      this._autocomplete.addListener('place_changed', this.locationChanged.bind(this))
+    }
+    return this._autocomplete
+  }
+
+  locationChanged() {
+    let place = this.autocomplete().getPlace()
+
+    if (!place.geometry) {
+      // User entered the name of a Place that was not suggested and
+      // pressed the Enter key, or the Place Details request failed.
+      window.alert("No details available for input: '" + place.name + "'");
+      return;
+    }
+
+    this.map().fitBounds(place.geometry.viewport)
+    this.map().setCenter(place.geometry.location)
+    this.marker().setPosition(place.geometry.location)
+    this.marker().setVisible(true)
+
+    this.latitudValue = place.geometry.location.lat()
+    this.longitudValue = place.geometry.location.lng()
+  }
+
+  preventSubmit(e) {
+    if (e.key == "Enter") { e.preventDefault() }
+  }
+
+  getCurrentLocation() {
+    console.log("FUNCIONA!")
```

Comment:
is this console log correct? 

Am I wondering if this method does not implement any logic what is the purpose to have it?

Reply:
Hi Ilyeo, I leave it because I'm going to implement that logic later. Also I leave the console.log to give my mates an example how stimulus controllers works

Reply:
alright then, thanks for the answer